contribs/drt/src/main/java/org/matsim/contrib/drt/optimizer/rebalancing/Feedforward/FeedforwardRebalancingStrategy.java

```diff
@@ -109,8 +113,15 @@ public FeedforwardRebalancingStrategy(DrtZonalSystem zonalSystem, Fleet fleet, R
 			relocationList.addAll(
 					fastHeuristicRelocationCalculator.calcRelocations(vehicleSurplusList, rebalancableVehiclesPerZone));
 		}
+
+		// Connection between feedback and feedforward part
 		Set<DvrpVehicle> relocatedVehicles = relocationList.stream().map(relocation -> relocation.vehicle)
 				.collect(toSet());
+		Map<DrtZone, List<DvrpVehicle>> actualRebalancableVehiclesPerZone = new HashMap<>();
+		for (DrtZone zone : rebalancableVehiclesPerZone.keySet()) {
+			actualRebalancableVehiclesPerZone.put(zone, rebalancableVehiclesPerZone.get(zone).stream()
+					.filter(v -> !relocatedVehicles.contains(v)).collect(Collectors.toList()));
+		}
```

Comment:
I think this whole section can be moved to inside the `if` structure, but outside the `for` loop, as we need it only if the feedforward is enabled.